### PR TITLE
Change importProjects to return a map of name -> project

### PR DESCRIPTION
**Pull request by briandealwis**
Change `ProjectUtils.importProject()` to return a map of _project-name_ → _project, since we have this information.  Simplifies `LocalAppEnginePublishOperationTest`.

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/project/ProjectUtils.java [hunk 1]
```diff
@@ -126,7 +128,7 @@
 
     // import the projects
     progress.setWorkRemaining(10 * projectFiles.size() + 10);
-    List<IProject> projects = new ArrayList<>(projectFiles.size());
+    Map<String, IProject> projects = Maps.newLinkedHashMap();
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/project/ProjectUtils.java]: Why not `new LinkedHashMap<>()`?

File: plugins/com.google.cloud.tools.eclipse.appengine.localserver.test/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEnginePublishOperationTest.java [hunk 2]
```diff
@@ -64,8 +64,8 @@ public void setUp() throws IOException, CoreException, InterruptedException {
     assertEquals(2, projects.size());
     Predicate<IProject> isServerProject = project -> "sox-server".equals(project.getName());
     Predicate<IProject> isSharedProject = project -> "sox-shared".equals(project.getName());
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.localserver.test/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEnginePublishOperationTest.java]: You can remove `isServerProject` and `isSharedProject` too.

File: plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/project/ProjectUtils.java [hunk 3]
```diff
@@ -126,7 +128,7 @@
 
     // import the projects
     progress.setWorkRemaining(10 * projectFiles.size() + 10);
-    List<IProject> projects = new ArrayList<>(projectFiles.size());
+    Map<String, IProject> projects = new LinkedHashMap<>();
     System.out.printf("Importing %d projects:\n", projectFiles.size());
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/project/ProjectUtils.java]: should we be using a logger here? or not printing at all?

File: plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/project/ProjectUtils.java [hunk 4]
```diff
@@ -126,7 +128,7 @@
...
     for (IPath projectFile : projectFiles) {
       System.out.println("    " + projectFile);
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/project/ProjectUtils.java]: ditto

File: plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/project/ProjectUtils.java (hunk 3 above)
Comment (briandealwis) [reply plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/project/ProjectUtils.java]: It's testing code, and so shows up on the Travis transcript.
```
Running com.google.cloud.tools.eclipse.appengine.compat.gpe.GpeClassicMigratorTest
Importing 1 projects:
    /home/travis/build/GoogleCloudPlatform/google-cloud-eclipse/plugins/com.google.cloud.tools.eclipse.appengine.compat.test/target/work/data/TaskQueueExamples/.project
Importing 1 projects:
    /home/travis/build/GoogleCloudPlatform/google-cloud-eclipse/plugins/com.google.cloud.tools.eclipse.appengine.compat.test/target/work/data/TaskQueueExamples/.project
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.866 sec - in com.google.cloud.tools.eclipse.appengine.compat.gpe.GpeClassicMigratorTest
```
Logging doesn't add any value here.

**Pull request merged**